docs/howto/shared-cache.rst

```diff
@@ -0,0 +1,73 @@
+.. _howto-shared-cache:
+
+Cache intermediate build artefacts
+==================================
+
+Because Charmcraft builds Python packages from source rather than using pre-built
+wheels, the initial builds of charms can take a while. However, future builds of
+the same charm are sped up significantly by reusing the built wheels. When installed
+as a snap, Charmcraft automatically caches these wheels in the
+``~/snap/charmcraft/common/cache`` directory. However, in some cases, it may be
+beneficial to change this directory. This can be done by setting the
+``CRAFT_SHARED_CACHE`` environment variable to the path of an existing directory to
+use instead.
+
+This can be especially useful in CI, where you may wish to specify a directory that
+gets cached between CI runs. This is done automatically when using the
+``charmcraft/pack`` action in Canonical's `craft-actions`_ repository for GitHub,
+but it can also be done manually in any CI.
+
+CI examples
+-----------
+
```

Comment:
We have a rule not to have two headings touching, so ideally we should add a sentence here.